src/tpm2_pytss/ESAPI.py

```diff
@@ -2179,6 +2179,65 @@ def hmac(
         )
         return TPM2B_DIGEST(_get_dptr(out_hMAC, lib.Esys_Free))
 
+    def mac(
+        self,
+        handle: ESYS_TR,
+        buffer: Union[TPM2B_MAX_BUFFER, bytes, str],
+        hash_alg: TPM2_ALG,
+        session1: ESYS_TR = ESYS_TR.PASSWORD,
+        session2: ESYS_TR = ESYS_TR.NONE,
+        session3: ESYS_TR = ESYS_TR.NONE,
+    ) -> TPM2B_DIGEST:
+        """Invoke the TPM2_MAC command.
+
+        This function invokes the TPM2_MAC command in a one-call
+        variant. This means the function will block until the TPM response is
+        available.
+
+        Args:
+            handle (ESYS_TR): Handle for the symmetric signing key providing the HMAC key.
+            buffer (Union[TPM2B_MAX_BUFFER, bytes, str]): MAC data.
+            hash_alg (TPM2_ALG): Algorithm to use for MAC.
+            session1 (ESYS_TR): A session for securing the TPM command (optional). Defaults to ESYS_TR.PASSWORD.
+            session2 (ESYS_TR): A session for securing the TPM command (optional). Defaults to ESYS_TR.NONE.
+            session3 (ESYS_TR): A session for securing the TPM command (optional). Defaults to ESYS_TR.NONE.
+
+        Raises:
+            TypeError: If a parameter is not of an expected type.
+            ValueError: If a parameter is not of an expected value.
+            TSS2_Exception: Any of the various TSS2_RC's the lower layers can return.
+
+        Returns:
+            A TPM2B_DIGEST result of the MAC.
+
+        C Function: Esys_MAC
+
+        TPM Command: TPM2_MAC
+        """
```

Comment:
I think we need to ensure that `lib.Esys_MAC` exists, because it will fail on older versions that don't have this.

Probably something like:
```python
if not hasattr(lib, "Esys_MAC"):
    raise NotImplementedError("ESAPI Does not have MAC Interface")
```

@whooo do you have a preferred way of doing this?

Reply:
Perhaps doing something similar to https://github.com/tpm2-software/tpm2-pytss/blob/master/src/tpm2_pytss/constants.py#L870 would be good, it would make to code look a bit weird but would make checking for support easier

Reply:
Yeah I like that better, using `if _lib_version_atleast("tss2-esapi", "3.0.0"):`

Reply:
Grr, this is where it gets hard, Since we're doing this ahead of tag 4.0, we need to check if attribute exists. We can switch this later....